### PR TITLE
Add Message Type and Fix Empty Message Bug, Closes Issues 11 and 13

### PR DESCRIPTION
**Pull request by jeramyRR**
Add Message Type (Closes Issue #13)
Fix empty message bug (Closes Issue #11)
Remove all spaces from NickName

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -49,26 +49,29 @@ func NewClient(name string, conn net.Conn, serverInput chan string) *Client {
...
+func (c *Client) processInput() {
 	for {
 		// The buffer will stop reading when it detects a new line chatacter
 		input, _ := c.reader.ReadString('\n')
 
-		// TODO: In the future, the server will be responsible for attaching
-		// the Client's NickName to the message
-		c.serverInput <- c.NickName + ": " + input
+		// If the client sends an empty message just ignore it
+		if input := trimMessage(input); len(input) > 0 {
```
Comment (ksolo) [inline client.go]: You may want to consider adding a helper/utility method to detect if the input is empty. I can see that check happening in other places and having a named abstraction for that could make the code more readable.

**Comment by ksolo** [comment]
I need to become more familiar with Go :)  I'll start going through a few tutorials and read up on it in the next few weeks to give more feedback.

File: client.go (hunk 1 above)
Comment (jeramyRR) [inline client.go]: Yeah, you're definitely right.  I'll add that to the issue tracker and try to get it in my next pull request.

Comment (ksolo) [inline client.go]: Instead of adding it as an issue, go ahead and fix is in this request. Once it's added, then we'll merge this in.

**Comment by jeramyRR** [comment]
Last push adds the helper/utility method

**Comment by ksolo** [comment]
:+1: 

Merging

**Pull request merged**